drivers/wifi/simplelink/simplelink_sockets.c

```diff
@@ -976,14 +977,36 @@ static void simplelink_freeaddrinfo(struct addrinfo *res)
 	free(res);
 }
 
-static int simplelink_fctnl(int fd, int cmd, va_list args)
+static int simplelink_fcntl(int sd, int cmd, va_list args)
 {
-	ARG_UNUSED(fd);
-	ARG_UNUSED(cmd);
-	ARG_UNUSED(args);
+	int retval = 0;
+	SlSockNonblocking_t enableOption;
+	SlSocklen_t optlen = sizeof(SlSockNonblocking_t);
+
+	switch (cmd) {
+	case F_GETFL:
+		sl_GetSockOpt(sd, SL_SOL_SOCKET, SL_SO_NONBLOCKING,
```

Comment:
Perhaps, here and below, check the return value of sl_[Get/Set]SockOpt(),  then pass retval via
`slcb_SetErrno(retval);`
as in the other cases.

Reply:
Thanks. I will make the change.